docs/topics/coroutines-overview.md

```diff
@@ -1,39 +1,126 @@
+
+To launch new coroutines, you can use coroutine builders like [`launch()`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html) and [`async()`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/async.html).
+These builders let you run tasks concurrently, return results, or enforce structured lifecycles.
+
+You can learn more about these builders in [Coroutine basics](coroutine-basics.md) and [Composing suspend functions](coroutines-and-channels.md).
+
+### Coroutine context and behavior
+
+Builder functions like `launch()` and `async()` automatically create a set of elements that define how the coroutine behaves:
+
+* [`Job`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-job/) interface: Tracks the coroutine’s lifecycle and enables structured concurrency.
+* [`CoroutineDispatcher`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-dispatcher/): Controls where the coroutine runs, such as on a background thread or the main thread in UI applications.
+* [`CoroutineExceptionHandler`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-exception-handler/): Handles uncaught exceptions.
+
+Together, these elements make up the [_coroutine context_](coroutine-context-and-dispatchers.md), which is inherited by default from the coroutine’s parent.
+This context forms a hierarchy that enables structured concurrency, where related coroutines can be [cancelled](cancellation-and-timeouts.md) together or [handle exceptions](exception-handling.md) as a group.
+
+### Asynchronous flow and shared mutable state
+
+Kotlin Coroutines provides the [`Flow`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-flow/) interface for working with asynchronous streams of values.
+While suspending functions return only a single result, flows let you emit and collect multiple values over time without blocking a thread.
+For more information, see [Asynchronous flow](flow.md)
+
+For direct communication between coroutines, use [`Channel`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-channel/), which allows one coroutine to send values to another.
+See [Channels](channels.md) for more information.
+
+When multiple coroutines need to access or update the same data, they _share mutable state_.
+Without coordination, this can lead to race conditions, where operations interfere with each other in unpredictable ways.
+To safely manage shared mutable state, the `kotlinx.coroutines` library provides constructs like [`Mutex`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.sync/-mutex/) and atomic variables.
+Learn more in [Shared mutable state and concurrency](shared-mutable-state-and-concurrency.md).
+
+## What's next
```

Comment:
The original page has a couple more links that seem important, but can, of course, be mentioned on the Basics page instead, WDYT?

* [Guide to UI programming with coroutines](https://github.com/Kotlin/kotlinx.coroutines/blob/master/ui/coroutines-guide-ui.md)
* [Full kotlinx.coroutines API reference](https://kotlinlang.org/api/kotlinx.coroutines/)